crates/ruff_python_ast/src/context.rs

```diff
@@ -203,6 +206,75 @@ impl<'a> Context<'a> {
         }
     }
 
+    /// Given a `module` and `member`, return the fully-qualified name of the binding in the current
+    /// scope, if it exists.
+    ///
+    /// E.g., given:
+    ///
+    /// ```python
+    /// from sys import version_info as python_version
+    /// print(python_version)
+    /// ```
+    ///
+    /// ...then `resolve_binding("sys", "version_info")` will return `Some("python_version")`.
+    pub fn resolve_binding(&self, module: &str, member: &str) -> Option<String> {
+        self.scopes().enumerate().find_map(|(scope_index, scope)| {
+            scope.binding_ids().find_map(|binding_index| {
+                match &self.bindings[*binding_index].kind {
+                    // Ex) Given `module="sys"` and `object="exit"`:
+                    // `import sys`         -> `sys.exit`
+                    // `import sys as sys2` -> `sys2.exit`
+                    BindingKind::Importation(Importation { name, full_name }) => {
+                        if full_name == &module {
+                            // Verify that `sys` isn't bound in an inner scope.
+                            if self
+                                .scopes()
+                                .take(scope_index)
+                                .all(|scope| scope.get(name).is_none())
+                            {
+                                return Some(format!("{name}.{member}"));
```

Comment:
This whole thing has the potential to be pretty slow.

We have to iterate over all bindings in scope (to find candidates), then iterate over the scopes we passed over to ensure that the candidates weren't overridden at a lower level. I'm sure we could make this much faster, but it'd require different bookkeeping.